lib/csp.js

```diff
@@ -4,15 +4,25 @@ const { v4: uuidv4 } = require('uuid')
 const CspStrategy = {}
 
 const defaultDirectives = {
-  defaultSrc: ['\'self\''],
-  scriptSrc: ['\'self\'', 'vimeo.com', 'https://gist.github.com', 'www.slideshare.net'],
-  imgSrc: ['*'],
-  styleSrc: ['\'self\'', '\'unsafe-inline\'', 'https://github.githubassets.com'], // unsafe-inline is required for some libs, plus used in views
-  fontSrc: ['\'self\'', 'data:', 'https://public.slidesharecdn.com'],
+  defaultSrc: ['\'none\''],
+  baseUri: ['\'self\''],
+  connectSrc: ['\'self\''],
+  fontSrc: ['\'self\''],
+  manifestSrc: ['\'self\''],
+  frameSrc: ['\'self\'', 'https://player.vimeo.com', 'https://www.slideshare.net/slideshow/embed_code/key/', 'https://www.youtube.com'],
+  imgSrc: ['*'], // we allow using arbitrary images
+  scriptSrc: [
+    config.serverURL + '/build/',
+    config.serverURL + '/js/',
+    config.serverURL + '/config',
+    'https://gist.github.com/',
+    'https://vimeo.com/api/oembed.json',
+    'https://www.slideshare.net/api/oembed/2',
+    '\'unsafe-inline\'' // this is ignored by browsers supporting nonces/hashes
```

Comment:
Uhm, this commend is not correct. `unsafe-inline` is not ignored, if unsafe-inline is there, one can use inline scripts.

> Inline script blocks MUST pass through § 4.2.3 Should element’s inline type behavior be blocked by Content Security Policy?. Their behavior will be blocked unless every policy allows inline script, either implicitly by not specifying a script-src (or default-src) directive, or explicitly, by specifying "unsafe-inline", a nonce-source or a hash-source that matches the inline block.

https://www.w3.org/TR/CSP3/#directive-script-src

Reply:
MDN writes:
>  Specifying nonce makes a modern browser ignore 'unsafe-inline' which could still be set for older browsers without nonce support. 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src

The linked CSP Scanner also does not complain about the presence of `unsafe-inline`:
> 'unsafe-inline' is ignored when a nonce or a hash is present on modern browsers, and serves for graceful degradation on legacy browsers.

Reply:
Well, I guess then I just learned something 👍🏻 

And apparently the standards are outdated again, one has to love this fast moving web.